Modules/xASL_module_ASL.m

```diff
@@ -101,14 +102,40 @@
 		x.Output_im = rmfield(x.Output_im,'ASL');
 	end
 
-	if isfield(x,'Output') && isfield(x.Output,'ASL')
-		x.Output = rmfield(x.Output,'ASL');
+	if isfield(x, 'Output') && isfield(x.Output, 'ASL') && isfield(x.Output.ASL, x.SESSION)
+		x.Output = rmfield(x.Output.ASL, x.SESSION);
 	end
 	
 	% And saved the cleaned up version
 	xASL_adm_SaveX(x);
+	
```

Comment:
Nice, but this should go to `xASL_adm_CleanUpBeforeRerun`, which is a separate function handling what needs to be removed for a rerun.

Reply:
That's how you wrote that previously. And I didn't want to touch that. I just made the necessary modifications. 

I can of course change that, but then please say here explicitly which changes should I do.

Reply:
It should not be too hard. The way I wrote xASL_adm_CleanUpBeforeRerun, is that it iterates over modules, such that you can call it for each module/session/subject separately. And if you clean a whole study (e.g., by calling it outside of xASL_Iteration) it will just loop/run over all those.

So the cleanup you want to add for a session, should go there under the ASL module/session part. It should be easy to follow that.

Reply:
Not so easy because that function doestn' have a x-struct input or output. But I have moved it to the other QC cleaning function in 4a9deb4f